server/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -72,16 +87,20 @@
     /**
      * Create a new translog checkpoint.
      *
-     * @param offset           the current offset in the translog
-     * @param numOps           the current number of operations in the translog
-     * @param generation       the current translog generation
-     * @param minSeqNo         the current minimum sequence number of all operations in the translog
-     * @param maxSeqNo         the current maximum sequence number of all operations in the translog
-     * @param globalCheckpoint the last-known global checkpoint
+     * @param offset                the current offset in the translog
+     * @param numOps                the current number of operations in the translog
+     * @param generation            the current translog generation
+     * @param minSeqNo              the current minimum sequence number of all operations in the translog
+     * @param maxSeqNo              the current maximum sequence number of all operations in the translog
+     * @param globalCheckpoint      the last-known global checkpoint
      * @param minTranslogGeneration the minimum generation referenced by the translog at this moment.
+     * @param trimmedAboveSeqNo     all operations with seq# above trimmedAboveSeqNo should be ignored and not read from the
```

Comment:
What do you mean ? To add overloaded constructor w/o trimmedAboveSeqNo argument ?

Reply:
Just from a documentation perspective. There is an ambiguity now since UNASSIGNED_SEQ_NO is -2, you may read this to mean that if you pass UNASSIGNED_SEQ_NO all ops will be trimmed.